lib/core/src/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobin.hs

```diff
@@ -1145,6 +1174,194 @@ makeChangeForCoin targets excess =
     weights :: NonEmpty Natural
     weights = coinToNatural <$> targets
 
+--------------------------------------------------------------------------------
+-- Equipartitioning
+--------------------------------------------------------------------------------
+
+-- An /equipartition/ of a value 'v' (of some type) is a /partition/ of that
+-- value into 'n' smaller values whose /sizes/ differ by no more than 1. The
+-- the notion of /size/ is dependent on the type of value 'v'.
+--
+-- In this section, equipartitions have the following properties:
+--
+--  1.  The length is observed:
+--      >>> length (equipartition v n) == n
+--
+--  2.  The sum is preserved:
+--      >>> sum (equipartition v n) == v
+--
+--  3.  Each resulting value is less than or equal to the original value:
+--      >>> all (`leq` v) (equipartition v n)
+--
+--  4.  The resultant list is sorted into ascending order when values are
+--      compared with the 'leq' function.
+--
+--------------------------------------------------------------------------------
+
+-- | Computes the equipartition of a coin into 'n' smaller coins.
+--
+equipartitionCoin
+    :: HasCallStack
+    => Coin
+    -- ^ The coin to be partitioned.
+    -> NonEmpty a
+    -- ^ Represents the number of portions in which to partition the coin.
+    -> NonEmpty Coin
+    -- ^ The partitioned coins.
+equipartitionCoin c =
```

Comment:
Given the max supply of ada, there's no need to apply this fix to Coin.

Reply:
> Given the max supply of ada, there's no need to apply this fix to Coin.

The point here is that we need to divide output coins together with maps, so that we can retain proportionality.

Let's say we need to partition the following token map into two:

- TokenMap A:
    - Banana: 10
    - Apple: 10

This would give us two token maps with the following tokens:

- TokenMap A1:
    - Banana: 5
    - Apple: 5
- TokenMap A2:
    - Banana: 5
    - Apple: 5

However, the change generation algorithm pairs each change token map with a coin from the original outputs, so that when we eventually assign ada values to change maps (to make change bundles), we can assign them in proportion to the sizes of the ada in the original outputs.

So we also need to divide the coin value.

Let's say this particular change map was paired with an output coin value of `100`. That will mean that each divided change map will be paired with a coin value of `50`.

For the sake of convenience, when dividing the pair (change map, output coin value), we treat it as a token bundle.